server/db_accessor.cpp

```diff
@@ -134,19 +132,52 @@ bool db_accessor::create_user(boost::asio::yield_context &yield, const std::stri
     return true;
 }
 
-std::optional<nibashared::character> nibaserver::db_accessor::get_char(const std::string &id) {
-    auto iter = char_tbl_.find(id);
-    if (iter == char_tbl_.end())
+// looks like this function 'broke' clang-format
+std::optional<nibashared::player>
+nibaserver::db_accessor::get_char(const std::string &id, boost::asio::yield_context &yield) {
+    BOOST_LOG_SEV(logger_, sev::info) << "fetching character for " << id;
+    ozo::error_code ec{};
+    // not the best, we can probably do better
+    // TODO: do something else, for example change how player is layed out
+    ozo::rows_of<std::string, char, int, int, int, int> characters;
+    auto conn = ozo::request(
+        conn_,
+        "SELECT name, gender, strength, dexterity, physique, spirit FROM player_character where id="_SQL +
+            id,
+        ozo::into(characters), yield[ec]);
+    if (ec) {
+        BOOST_LOG_SEV(logger_, sev::error) << ec.message() << " | " << ozo::error_message(conn)
+                                           << " | " << ozo::get_error_context(conn);
+        return {};
+    }
+    if (characters.size() == 0)
         return {};
-    return iter->second;
+    auto &[name, gender, strength, dexterity, physique, spirit] = characters.at(0);
+    return nibashared::player{
+        .name = name,
+        .gender = gender,
+        .attrs = {
+            .strength = strength, .dexterity = dexterity, .physique = physique, .spirit = spirit}};
 }
 
-bool db_accessor::create_char(const std::string &id, nibashared::character &&character) {
-    auto c = get_char(id);
-    if (c) {
+bool db_accessor::create_char(const std::string &id, const nibashared::player &player,
+                              boost::asio::yield_context &yield) {
+    ozo::error_code ec{};
+    // the insert string is unfortunately long and hand written
```

Comment:
Static data will be longer.

Reply:
there are json dump views set up, you can leverage that